src/plugins/usage_collection/server/routes/report_metrics.ts

```diff
@@ -48,15 +53,97 @@ export function registerUiMetricRoute(
+          }; // reseting the batchReport and updating the startTimestamp to current TimeStamp
+
+          if (prevReport) {
+            // Write the previously batched Report to the saved object
+            const internalRepository = getSavedObjects();
+            if (!internalRepository) {
+              throw Error(`The saved objects client hasn't been initialised yet`);
+            }
+            await storeReport(internalRepository, prevReport.report);
+          }
         }
-        await storeReport(internalRepository, report);
+
         return res.ok({ body: { status: 'ok' } });
       } catch (error) {
         return res.ok({ body: { status: 'fail' } });
       }
     }
   );
 }
+
+function combineReports(report1: ReportSchemaType, report2: ReportSchemaType) {
+  // Combines report2 onto the report1 and returns the updated report1
+
+  // Combining User Agents
+  const combinedUserAgent = { ...report1.userAgent };
+
+  if (report2.userAgent !== undefined) {
+    for (const key in Object.keys(report2.userAgent)) {
+      if (report2.userAgent[key] === undefined) {
+        continue;
+      }
+
+      // Only adding the useragents not already present in report1
+      if (!report1.userAgent || report1.userAgent[key] === undefined) {
+        combinedUserAgent[key] = report2.userAgent[key];
+      }
+    }
+  }
+
+  // Combining UI metrics
+  const combinedUIMetric = { ...report1.uiStatsMetrics };
+  if (report2.uiStatsMetrics !== undefined) {
+    for (const key of Object.keys(report2.uiStatsMetrics)) {
+      if (report2.uiStatsMetrics[key] === undefined) {
+        continue;
+      } else if (
+        report1.uiStatsMetrics === undefined ||
+        report1.uiStatsMetrics[key] === undefined
+      ) {
+        combinedUIMetric[key] = report2.uiStatsMetrics[key];
+      } else {
+        const { stats, ...rest } = combinedUIMetric[key];
+        const combinedStats = { ...stats };
+        combinedStats.sum += report2.uiStatsMetrics[key].stats.sum; // Updating the sum since it is field we will be using to update the saved Object
+        combinedUIMetric[key] = { ...rest, stats: combinedStats };
+      }
+    }
+  }
+
+  // Combining Application Usage
+  const combinedApplicationUsage = { ...report1.application_usage };
+  if (report2.application_usage !== undefined) {
+    for (const key in Object.keys(report2.application_usage)) {
+      if (report2.application_usage[key] === undefined) {
+        continue;
+      } else if (!report1.application_usage || report1.application_usage[key] === undefined) {
+        combinedApplicationUsage[key] = report2.application_usage[key];
+      } else {
+        const combinedUsage = { ...combinedApplicationUsage[key] };
+        combinedUsage.numberOfClicks += report2.application_usage[key].numberOfClicks;
+        combinedUsage.minutesOnScreen += report2.application_usage[key].minutesOnScreen;
+        combinedApplicationUsage[key] = combinedUsage;
+      }
+    }
```

Comment:
```suggestion
    for (const key of Object.keys(report2.application_usage)) {
      if (report2.application_usage[key] === undefined) continue;
      
      if (report1.application_usage?.[key] === undefined) {
        combinedApplicationUsage[key] = report2.application_usage[key];
      } else {
        combinedApplicationUsage[key].numberOfClicks += report2.application_usage[key].numberOfClicks || 0;
        combinedApplicationUsage[key].minutesOnScreen += report2.application_usage[key].minutesOnScreen || 0;
      }
    }
```

Reply:
Also, don't we need to guard against missing `numberOfClicks` and `minutesOnScreen` in `report2.application_usage[key]`?

Reply:
Thanks for pointing it out. I have updated the check conditions to guard against missing `numberOfClicks` and `minutesOnScreen `.